pkg/udf/map_udf.go

```diff
@@ -54,18 +55,28 @@ func (u *MapUDFProcessor) Start(ctx context.Context) error {
 		return fmt.Errorf("unrecognized isbsvc type %q", u.ISBSvcType)
 	}
 
+	// Populate shuffle function map
+	shuffleFuncMap := make(map[string]*shuffle.Shuffle)
+	for _, edge := range u.VertexInstance.Vertex.Spec.ToEdges {
+		if edge.Parallelism != nil && *edge.Parallelism > 1 {
+			s := shuffle.NewShuffle(dfv1.GenerateEdgeBufferNames(u.VertexInstance.Vertex.Namespace, u.VertexInstance.Vertex.Spec.PipelineName, edge))
+			shuffleFuncMap[fmt.Sprintf("%s:%s", edge.From, edge.To)] = s
+		}
+	}
+
 	conditionalForwarder := forward.GoWhere(func(key string) ([]string, error) {
 		result := []string{}
-		_key := string(key)
-		if _key == dfv1.MessageKeyAll || _key == dfv1.MessageKeyDrop {
-			result = append(result, _key)
+		if key == dfv1.MessageKeyDrop {
 			return result, nil
 		}
-		for _, to := range u.VertexInstance.Vertex.Spec.ToEdges {
-			// If returned key is not "ALL" or "DROP", and there's no conditions defined in the edge,
-			// treat it as "ALL"?
-			if to.Conditions == nil || len(to.Conditions.KeyIn) == 0 || sharedutil.StringSliceContains(to.Conditions.KeyIn, _key) {
-				result = append(result, dfv1.GenerateEdgeBufferNames(u.VertexInstance.Vertex.Namespace, u.VertexInstance.Vertex.Spec.PipelineName, to)...)
+		for _, edge := range u.VertexInstance.Vertex.Spec.ToEdges {
```

Comment:
How do we tackle reduce's ISBs v/s the map's ISB? won't the `ToEdges` have both kinds of edges, multiple edges for the reduce vertex and a single edge for the map?

Reply:
Edge is not buffer. No matter it is reduce vertex or map vertex, there's only one edge for them. However, reduce edge may have multiple buffers, map edge only has one.